plugins/action/zos_job_submit.py

```diff
@@ -60,6 +60,11 @@ def run(self, tmp=None, task_vars=None):
             dest_path = self._execute_module(
                 module_name="tempfile", module_args={}, task_vars=task_vars,
             ).get("path")
+            # Calling execute_module from this step with tempfile leaves behind a tmpdir.
+            # This is called to ensure the proper removal.
+            tmpdir = self._connection._shell.tmpdir
+            if tmpdir:
+                self._remove_tmp_path(tmpdir)
```

Comment:
So would this remove the entire `tmpdir`? Would this affect other files used by Ansible or the collection?

Reply:
Not really here is a fragment of the module execution : 
```
<EC01135A.vmec.svl.ibm.com> (0, b'ansible-tmp-1700775014.894572-16119-229678277378752=/.ansible/tmp/ansible-tmp-1700775014.894572-16119-229678277378752\n', b'')
Using module file /Users/fernandoflores/Documents/IBM/venv/AC2.14PY3.9/lib/python3.9/site-packages/ansible/modules/tempfile.py
<EC01135A.vmec.svl.ibm.com> PUT /Users/fernandoflores/.ansible/tmp/ansible-local-16116shjwub4o/tmpewrusozf TO /.ansible/tmp/ansible-tmp-1700775014.894572-16119-229678277378752/AnsiballZ_tempfile.py
```

At this step tmpdir of the connection shell is `/.ansible/tmp/ansible-tmp-1700775014.894572-16119-229678277378752/` but later on the execution is changed to `/.ansible/tmp/ansible-tmp-1700775020.609182-16119-274785560197872`. strangely enough, this is the only module I found that keeps the AnsiballZ file from `tempfile` execution, zos_unarchive and zos_copy use that too but gets removed.

See logs attached 
[test.log](https://github.com/ansible-collections/ibm_zos_core/files/13475954/test.log)

Reply:
I see, you're removing the temp directory created for the task, thanks!